### PR TITLE
RDBQA-82 - studio changes to allow encrypted databases over http

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RDBQA-82

### Additional description

Studio changes to allow encrypted databases over HTTP.

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [x] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/commands/resources/getGlobalStudioConfigurationCommand.ts [hunk 1]
```diff
@@ -20,6 +28,10 @@ class getGlobalStudioConfigurationCommand extends commandBase {
         
         return loadTask;
     }
+
+    private processHeaders(response: JQueryXHR) {
+        accessManager.default.allowEncryptedDatabasesOverHttp(response.getResponseHeader("AllowEncryptedDatabasesOverHttp") == "true");
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/resources/getGlobalStudioConfigurationCommand.ts]: `===`

File: src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html [hunk 2]
```diff
@@ -58,11 +58,11 @@ <h3>NAME<i class="required"></i></h3>
                             </div>
                         </div>
                         <div>
-                           <div class="encryption" data-bind="template: { name: 'create-db-encryption' }, visible: advancedVisibility.encryption() && isSecureServer">
+                           <div class="encryption" data-bind="template: { name: 'create-db-encryption' }, visible: advancedVisibility.encryption() && (isSecureServer || allowEncryptedDatabasesOverHttp)">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html]:
```suggestion
                           <div class="encryption" data-bind="template: { name: 'create-db-encryption' }, visible: advancedVisibility.encryption() && (isSecureServer || allowEncryptedDatabasesOverHttp())">
```

File: src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html [hunk 3]
```diff
@@ -58,11 +58,11 @@ <h3>NAME<i class="required"></i></h3>
...
                            </div>
                         </div>
                         <div>
-                            <div class="encryption" data-bind="template: { name: 'authentication-is-off' }, visible: advancedVisibility.encryption() && !isSecureServer">
+                            <div class="encryption" data-bind="template: { name: 'authentication-is-off' }, visible: advancedVisibility.encryption() && (!isSecureServer && !allowEncryptedDatabasesOverHttp)">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html]:
```suggestion
                            <div class="encryption" data-bind="template: { name: 'authentication-is-off' }, visible: advancedVisibility.encryption() && (!isSecureServer && !allowEncryptedDatabasesOverHttp())">
```

File: src/Raven.Studio/typescript/commands/resources/getGlobalStudioConfigurationCommand.ts (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Studio/typescript/commands/resources/getGlobalStudioConfigurationCommand.ts]: done

File: src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html]: @ml054 I also need `isSecureServer` -> `isSecureServer()`?

File: src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html]: done

**Review by ml054 — APPROVED** [review]

**Comment by ppekrol** [comment]
This requires separate PR to 6.0

**Pull request merged**